src/states/elder/mod.rs

```diff
@@ -710,7 +727,91 @@ impl Elder {
             return;
         }
 
-        self.vote_for_event(AccumulatingEvent::Online(pub_id));
+        // This joining node is being relocated to us.
+        let age = if let Some(payload) = relocate_payload {
+            if !payload.verify_identity(&pub_id) {
+                debug!(
+                    "{} - Ignoring relocation JoinRequest from {} - invalid signature.",
+                    self, pub_id
+                );
+                return;
+            }
+
+            let details = payload.details;
+
+            if !self
+                .chain
+                .our_prefix()
+                .matches(&details.content().destination)
+            {
+                debug!("{} - Ignoring relocation JoinRequest from {} - destination {} doesn't match our prefix {:?}.",
+                    self, pub_id, details.content().destination, self.chain.our_prefix());
+                return;
+            }
+
+            let age = details.content().age;
+            let message = SignedRoutingMessage::from(details);
+
+            if let Err(err) = message.check_integrity() {
+                debug!("{} - Ignoring relocation JoinRequest from {} - invalid integrity of {:?}: {:?}.", self, pub_id, message, err);
```

Comment:
this line seems too long, wondering why it not being complained by the format checker?

Reply:
No idea. I'll reformat it manually this time, although I generally prefer not to fight with rustfmt.

Reply:
normally there shall be a `line exceeding` error from rustfmt. :(

Reply:
I haven't seen that error for some time so I don't think rustfmt does that anymore. At least not by default.